h5grove/models.py

```diff
@@ -40,3 +40,54 @@ class LinkResolution(str, Enum):
     },
     total=False,
 )
+
+
+class EntityMetadata(TypedDict):
+    name: str
+    kind: str
+
+
+class ExternalLinkMetadata(EntityMetadata):
+    target_file: str
+    target_path: str
+
+
+class SoftLinkMetadata(EntityMetadata):
+    target_path: str
+
+
+AttributeMetadata = TypedDict(
+    "AttributeMetadata", {"name": str, "shape": tuple, "type": TypeMetadata}
+)
+
+
+class ResolvedEntityMetadata(EntityMetadata):
+    attributes: List[AttributeMetadata]
+
+
+class GroupMetadata(ResolvedEntityMetadata):
+    children: NotRequired[List[EntityMetadata]]
+
+
+class DatasetMetadata(ResolvedEntityMetadata):
+    chunks: tuple
+    filters: tuple
+    shape: tuple
+    type: TypeMetadata
+
+
+class DatatypeMetadata(ResolvedEntityMetadata):
+    type: TypeMetadata
+
+
+Stats = TypedDict(
+    "Stats",
+    {
+        "strict_positive_min": Optional[Union[int, float]],
+        "positive_min": Optional[Union[int, float]],
+        "min": Optional[Union[int, float]],
+        "max": Optional[Union[int, float]],
+        "mean": Optional[Union[int, float]],
+        "std": Optional[Union[int, float]],
```

Comment:
```suggestion
        "strict_positive_min": int | float | None,
        "positive_min": int | float | None,
        "min": int | float | None,
        "max": int | float | None,
        "mean": int | float | None,
        "std": int | float | None,
```

Reply:
` | ` didn't seem to work here.

Reply:
Indeed.
Sorry, I've been too fast, :coffee: was pending.

It's possible to use `|` etc... when it is a "type annotation" because with `from __future__ import annotations`, "type annotations" are lazily evaluated.
Here the type needs to be evaluated to instantiate the `dict`, so no choice to keep `Union`